File: GFG/Segregate0and1/Segregate0and1.js
/**
* @param {number[]} arr
*/
class Solution {
	segregate0and1(arr) {
		// code here
		let left = 0;
		let right = arr.length - 1;
		while (left < right) {
		    if (arr[left] === 0)  {
		        left++;
		    } else {
		        [arr[left], arr[right]] = [arr[right], arr[left]];
		        right--;
		    }
		}
	}
}

File: GFG/Segregate0and1/Segregate0and1.cpp
class Solution {
  public:
    void segregate0and1(vector<int> &arr) {
		// code here
		int left = 0;
		int right = arr.size() - 1;
		while (left < right) {
		    if (arr[left] == 0)  {
		        left++;
		    } else {
		        swap(arr[left], arr[right]);
		        right--;
		    }
		}
	}
};
